src/shell/interpreter.zig

```diff
@@ -9456,6 +9437,192 @@ pub const Interpreter = struct {
+                        continue;
+                    }
+                    if (std.mem.startsWith(u8, arg, "-t")) {
+                        this.terminator = arg[2..];
+                        continue;
+                    }
+
+                    if (std.mem.eql(u8, arg, "-w") or std.mem.eql(u8, arg, "--fixed-width")) {
+                        this.fixed_width = true;
+                        continue;
+                    }
+
+                    iter.index -= 1;
+                    break;
+                }
+
+                const maybe1 = iter.next().?;
+                const int1 = bun.fmt.parseFloat(f32, bun.sliceTo(maybe1, 0)) catch return this.fail("seq: invalid argument\n");
+                this.end = int1;
+                if (this.start > this.end) this.increment = -1;
+
+                const maybe2 = iter.next();
+                if (maybe2 == null) return this.do();
+                const int2 = bun.fmt.parseFloat(f32, bun.sliceTo(maybe2.?, 0)) catch return this.fail("seq: invalid argument\n");
+                this.start = int1;
+                this.end = int2;
+                if (this.start < this.end) this.increment = 1;
+                if (this.start > this.end) this.increment = -1;
+
+                const maybe3 = iter.next();
+                if (maybe3 == null) return this.do();
+                const int3 = bun.fmt.parseFloat(f32, bun.sliceTo(maybe3.?, 0)) catch return this.fail("seq: invalid argument\n");
+                this.start = int1;
+                this.increment = int2;
+                this.end = int3;
+
+                if (this.increment == 0) return this.fail("seq: zero increment\n");
+                if (this.start > this.end and this.increment > 0) return this.fail("seq: needs negative decrement\n");
+                if (this.start < this.end and this.increment < 0) return this.fail("seq: needs positive increment\n");
+
+                return this.do();
+            }
+
+            fn fail(this: *@This(), msg: string) Maybe(void) {
+                if (this.bltn.stderr.needsIO()) {
+                    this.state = .err;
+                    this.bltn.stderr.enqueue(this, msg);
+                    return Maybe(void).success;
+                }
+                _ = this.bltn.writeNoIO(.stderr, msg);
+                this.bltn.done(1);
+                return Maybe(void).success;
+            }
+
+            fn do(this: *@This()) Maybe(void) {
+                var current = this.start;
+                var arena = std.heap.ArenaAllocator.init(bun.default_allocator);
+                defer arena.deinit();
+
+                while (if (this.increment > 0) current <= this.end else current >= this.end) : (current += this.increment) {
+                    const str = std.fmt.allocPrint(arena.allocator(), "{d}", .{current}) catch bun.outOfMemory();
+                    defer _ = arena.reset(.retain_capacity);
```

Comment:
What about printing to `this.buf.writer()` instead?

Reply:
`buf` is only used if `this.bltn.stdout.needsIO()`